Sprint 8 HomeWork.ver1

DmitryBrosalin opened the pull request:
Добавил дату и время старта задачи и продолжительность. Добавил функцию вывода задач в порядке приоритета. Добавил проверку пересечений по времени.
Добавил новые тесты.

DmitryBrosalin committed:
Sprint 8 HomeWork.ver1
Добавил дату и время старта задачи и продолжительность.
Добавил функцию вывода задач в порядке приоритета.
Добавил проверку пересечений по времени.
Добавил новые тесты.

DmitryBrosalin committed:
Sprint 8 HomeWork.ver1
Добавил дату и время старта задачи и продолжительность.
Добавил функцию вывода задач в порядке приоритета.
Добавил проверку пересечений по времени.
Добавил новые тесты.

DmitryBrosalin committed:
Sprint 8 HomeWork.ver1
Добавил дату и время старта задачи и продолжительность.
Добавил функцию вывода задач в порядке приоритета.
Добавил проверку пересечений по времени.
Добавил новые тесты.

slavrikk commented on src/managers/InMemoryTaskManager.java:
```diff
@@ -180,4 +237,19 @@ public List<Task> getHistory() {
+
+    @Override
+    public TreeSet<Task> getPrioritizedTasks() {
+        return prioritizedTasks;
```
Тут лучше вернуть копию

slavrikk commented on src/managers/InMemoryTaskManager.java:
```diff
@@ -120,35 +134,55 @@ public void addNewEpic(Epic epic) {
+            Subtask oldSubtask = subtasks.get(subtask.getId());
+            Epic oldParentEpic = epics.get(oldSubtask.getParentEpicID());
+            oldParentEpic.getSubtasksID().remove((Object) subtask.getId());
+            prioritizedTasks.remove(subtasks.get(subtask.getId()));
```
На 170 строке уже выполнилось удаление

slavrikk commented on src/managers/InMemoryTaskManager.java:
```diff
@@ -180,4 +237,19 @@ public List<Task> getHistory() {
+    }
+
+    @Override
+    public boolean causesTimeConflict(Task t1, Task t2) {
```
Этот метод лучше сделать приватным, так как это внутренняя реализация класса

DmitryBrosalin replied:
У меня не получается сделать его приватным, потому что:
1) он прописан в интерфейсе TaskManager - idea не дает мне записать его private, пишет так:
```
'causesTimeConflict(Task, Task)' in 'managers.InMemory TaskManager' clashes with 'causesTimeConflict (Task, Task)' in 'managers.
TaskManager'; attempting to assign weaker access privileges ('private'); was 'public'
```
2) при private не получится его использовать в дочернем классе FileBackedTaskManager

Если есть какая-то уловка, чтобы реализовать модификатор доступа private, прошу о подсказке

slavrikk replied:
1. Убрать метод causesTimeConflict из интерфейса
2. Если метод используется в FileBackedTaskManager, тогда можно оставить protected

slavrikk commented on src/managers/InMemoryTaskManager.java:
```diff
@@ -171,6 +204,30 @@ public void checkEpicState(Epic epic) {
     }
 
+    @Override
+    public void checkEpicTime(Epic epic) {
```
Этот метод лучше сделать приватным, так как это внутренняя реализация класса

DmitryBrosalin replied:
та же самая проблема, что и с предыдущим комментарием

slavrikk replied:
Аналогичный комментарий

slavrikk commented on src/taskclasses/Task.java:
```diff
@@ -1,15 +1,38 @@
+import java.time.format.DateTimeFormatter;
+
 public class Task {
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm dd.MM.yy");
```
Можно указать private

DmitryBrosalin replied:
максимум могу protected, так как он используется в методе toString дочернего класса Subtask

slavrikk replied:
Согласен, тогда лучше protected

slavrikk commented on test/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,268 @@
+        assertEquals(epics, savedEpics, "Ошибка при возврате эпиков");
+        assertEquals(subtasks, savedSubtasks, "Ошибка при возврате подзадач");
+        assertFalse(savedTasks.containsValue(task0), "Ошибка при расчете конфликта задач по времени");
+        file.delete();
```
Удаление можно вынести в метод `@AfterEach`

slavrikk commented on .idea/misc.xml:
```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Папку .idea следует удалить из удаленного репозитория

DmitryBrosalin committed:
Delete .idea directory

DmitryBrosalin committed:
Sprint 8 HomeWork.ver2
- Изменил модификаторы доступа на protected в методах:
checkEpicTime
checkEpicState
causesTimeConflict
- getPrioritizedTasks теперь возвращает копию
- Изменил модификатор доступа на protected у поля DateTimeFormatter
- Добавил удаление файла в тестах через @AfterEach

DmitryBrosalin committed:
Sprint 8 HomeWork.ver2
- Изменил модификаторы доступа на protected в методах:
checkEpicTime
checkEpicState
causesTimeConflict
- getPrioritizedTasks теперь возвращает копию
- Изменил модификатор доступа на protected у поля DateTimeFormatter
- Добавил удаление файла в тестах через @AfterEach

DmitryBrosalin committed:
Merge remote-tracking branch 'origin/sprint_8-solution-time-and-duration' into sprint_8-solution-time-and-duration
